### PR TITLE
Add fix for process and include details on crash behavior.

### LINKED ISSUE #5855: panic: runtime error: in prometheus.exporter.windows
### What's wrong?

Flow Agent with "prometheus.exporter.windows" with process collector enabled crashes at startup.

### Steps to reproduce

1. Start a grafana-agent on windows 10
2. Agent crashes with these logs:

### System information

Windows 10

### Software version

agent, version v0.38.0 build date:       2023-11-21T14:07:41Z

### Configuration

```text
logging {
	level = "debug"
}


prometheus.exporter.windows "local" {
	enabled_collectors	= 	["cpu","cs","logical_disk","net","os","system","memory", "process"]
	logical_disk {
		include = "(C:|D:)"
	}
	
	network  {
		include = ".*"
	}
	
	process  {
		include = "java.*"
	}
}

prometheus.scrape "remote" {

  scrape_timeout = "2s"
  scrape_interval = "2s"
  targets    = prometheus.exporter.windows.local.targets
  forward_to = [prometheus.remote_write.push.receiver]
}

prometheus.remote_write "push" {
  endpoint {
    url = "http://localhost:9090/api/v1/write"
  }
}
```

### Logs

```text
C:\Program Files\Grafana Agent Flow>grafana-agent-flow-windows-amd64.exe run config.river
ts=2023-11-27T00:07:15.9031063Z level=info "boringcrypto enabled"=false
ts=2023-11-27T00:07:15.9041272Z level=info msg="running usage stats reporter"
ts=2023-11-27T00:07:15.9041272Z level=info msg="starting complete graph evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31
ts=2023-11-27T00:07:15.9041272Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=tracing duration=0s
ts=2023-11-27T00:07:15.9041272Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=labelstore duration=0s
ts=2023-11-27T00:07:15.9080822Z level=info msg="replaying WAL, this may take a while" component=prometheus.remote_write.push subcomponent=wal dir=data-agent\prometheus.remote_write.push\wal
ts=2023-11-27T00:07:15.9080822Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=0 maxSegment=20
ts=2023-11-27T00:07:15.9090792Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=1 maxSegment=20
ts=2023-11-27T00:07:15.9205717Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=2 maxSegment=20
ts=2023-11-27T00:07:15.9205717Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=3 maxSegment=20
ts=2023-11-27T00:07:15.921569Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=4 maxSegment=20
ts=2023-11-27T00:07:16.0052906Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=5 maxSegment=20
ts=2023-11-27T00:07:16.0373967Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=6 maxSegment=20
ts=2023-11-27T00:07:16.0375276Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=7 maxSegment=20
ts=2023-11-27T00:07:16.0385283Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=8 maxSegment=20
ts=2023-11-27T00:07:16.0395256Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=9 maxSegment=20
ts=2023-11-27T00:07:16.0405198Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=10 maxSegment=20
ts=2023-11-27T00:07:16.0425239Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=11 maxSegment=20
ts=2023-11-27T00:07:16.0495256Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=12 maxSegment=20
ts=2023-11-27T00:07:16.0495256Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=13 maxSegment=20
ts=2023-11-27T00:07:16.0614513Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=14 maxSegment=20
ts=2023-11-27T00:07:16.0614513Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=15 maxSegment=20
ts=2023-11-27T00:07:16.0614513Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=16 maxSegment=20
ts=2023-11-27T00:07:16.0614513Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=17 maxSegment=20
ts=2023-11-27T00:07:16.0634433Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=18 maxSegment=20
ts=2023-11-27T00:07:16.0634433Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=19 maxSegment=20
ts=2023-11-27T00:07:16.0644416Z level=info msg="WAL segment loaded" component=prometheus.remote_write.push subcomponent=wal segment=20 maxSegment=20
ts=2023-11-27T00:07:16.0644416Z level=info msg="Starting WAL watcher" component=prometheus.remote_write.push subcomponent=rw remote_name=96c840 url=http://localhost:9090/api/v1/write queue=96c840
ts=2023-11-27T00:07:16.0644416Z level=info msg="Starting scraped metadata watcher" component=prometheus.remote_write.push subcomponent=rw remote_name=96c840 url=http://localhost:9090/api/v1/write
ts=2023-11-27T00:07:16.0654376Z level=info msg="Replaying WAL" component=prometheus.remote_write.push subcomponent=rw remote_name=96c840 url=http://localhost:9090/api/v1/write queue=96c840
ts=2023-11-27T00:07:16.0654376Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=prometheus.remote_write.push duration=161.3104ms
ts=2023-11-27T00:07:16.0654376Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=otel duration=0s
ts=2023-11-27T00:07:16.0654376Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=logging duration=0s
ts=2023-11-27T00:07:16.0654376Z level=info msg="applying non-TLS config to HTTP server" service=http
ts=2023-11-27T00:07:16.0664343Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=http duration=996.7µs
ts=2023-11-27T00:07:16.0664343Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=cluster duration=0s
ts=2023-11-27T00:07:16.0664343Z level=info msg="enabled windows_exporter collectors" component=prometheus.exporter.windows.local collectors=cpu,cs,logical_disk,memory,net,os,process,system
ts=2023-11-27T00:07:16.0664343Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=prometheus.exporter.windows.local duration=0s
ts=2023-11-27T00:07:16.929239Z level=debug msg="scrape config was updated" component=prometheus.scrape.remote
ts=2023-11-27T00:07:16.929239Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=prometheus.scrape.remote duration=862.8047ms
ts=2023-11-27T00:07:16.9303589Z level=info msg="finished node evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 node_id=ui duration=0s
ts=2023-11-27T00:07:16.9309136Z level=info msg="finished complete graph evaluation" controller_id="" trace_id=872a07d0e80c1072936dacc51fee4d31 duration=1.0267864s
ts=2023-11-27T00:07:16.9316007Z level=debug msg="changing node state" from=viewer to=participant
ts=2023-11-27T00:07:16.9316007Z level=info msg="scheduling loaded components and services"
ts=2023-11-27T00:07:16.9316007Z level=info msg="finished node evaluation" controller_id="" node_id=prometheus.scrape.remote duration=0s
ts=2023-11-27T00:07:16.9316007Z level=debug msg="FANTB1 @1: participant"
ts=2023-11-27T00:07:16.9326217Z level=debug msg="passed new targets to scrape manager" component=prometheus.scrape.remote
ts=2023-11-27T00:07:16.9326379Z level=info msg="starting cluster node" peers="" advertise_addr=127.0.0.1:12345
ts=2023-11-27T00:07:16.9326379Z level=info msg="finished node evaluation" controller_id="" node_id=prometheus.scrape.remote duration=0s
ts=2023-11-27T00:07:16.9331418Z level=info msg="now listening for http traffic" service=http addr=127.0.0.1:12345
ts=2023-11-27T00:07:16.9337805Z level=debug msg="FANTB1 @3: participant"
ts=2023-11-27T00:07:16.9342946Z level=info msg="peers changed" new_peers=FANTB1
ts=2023-11-27T00:07:22.9351147Z level=debug msg="dialed to in-memory listener" service=http
ts=2023-11-27T00:07:22.9351147Z level=debug msg="accepted new in-memory connection" service=http
ts=2023-11-27T00:07:24.1467458Z level=debug msg="collector net succeeded after 0.000000s." component=prometheus.exporter.windows.local
ts=2023-11-27T00:07:24.1467458Z level=debug msg="collector system succeeded after 0.000000s." component=prometheus.exporter.windows.local
ts=2023-11-27T00:07:24.147258Z level=debug msg="collector memory succeeded after 0.000512s." component=prometheus.exporter.windows.local
ts=2023-11-27T00:07:24.147258Z level=debug msg="collector cs succeeded after 0.000512s." component=prometheus.exporter.windows.local
ts=2023-11-27T00:07:24.147258Z level=debug msg="collector logical_disk succeeded after 0.000512s." component=prometheus.exporter.windows.local
ts=2023-11-27T00:07:24.147258Z level=debug msg="collector cpu succeeded after 0.000512s." component=prometheus.exporter.windows.local
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x56bd355]

goroutine 405 [running]:
github.com/prometheus-community/windows_exporter/pkg/collector/process.(*collector).Collect(0xc0025cedd0, 0xc0035fb168, 0x0?)
        /go/pkg/mod/github.com/prometheus-community/windows_exporter@v0.24.1-0.20231116150933-ab05f43716d4/pkg/collector/process/process.go:309 +0xd5
github.com/prometheus-community/windows_exporter/pkg/collector.(*Prometheus).execute(0xc0025ed800, {0x7b4187f, 0x7}, {0x8c02dc0, 0xc0025cedd0}, 0x0?, 0x1822?)
        /go/pkg/mod/github.com/prometheus-community/windows_exporter@v0.24.1-0.20231116150933-ab05f43716d4/pkg/collector/prometheus.go:176 +0x8f
github.com/prometheus-community/windows_exporter/pkg/collector.(*Prometheus).Collect.func2({0x7b4187f, 0x7}, {0x8c02dc0?, 0xc0025cedd0?})
        /go/pkg/mod/github.com/prometheus-community/windows_exporter@v0.24.1-0.20231116150933-ab05f43716d4/pkg/collector/prometheus.go:117 +0xa5
created by github.com/prometheus-community/windows_exporter/pkg/collector.(*Prometheus).Collect in goroutine 364
        /go/pkg/mod/github.com/prometheus-community/windows_exporter@v0.24.1-0.20231116150933-ab05f43716d4/pkg/collector/prometheus.go:115 +0x43b
```

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

This updates windows_exporter to a version that handles the process config correctly and no longer lets a nil slip through.

#### Which issue(s) this PR fixes

Fixes #5855

#### Notes to the Reviewer

Ran with the default configuration for all  the available collectors except for the ones mentioned in the admonition since those caused panics when calling underlying windows layer.

#### PR Checklist
 
Did not update the changelog since it falls under my previous change of fixing defaults.

- [X] Documentation added

### CODE REVIEW

**Comment by mattdurham** [comment]
@clayton-cornell is this the right use of the admonition note?

File: docs/sources/flow/reference/components/prometheus.exporter.windows.md [hunk 1]
```diff
@@ -271,6 +271,11 @@ Name     | Description | Enabled by default
 
 See the linked documentation on each collector for more information on reported metrics, configuration settings and usage examples.
 
+{{% admonition type="caution" %}}
+Certain collectors will cause the application to crash if they are used and the required infrastructure is not installed.
+These include but are not limited to mscluster_*, vmware, nps, dns, msmq, teradici_pcoip, ad, hyperv, and scheduled_task.
+{{%/admonition %}}
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.windows.md:277]:
```suggestion
{{% /admonition %}}
```

File: docs/sources/flow/reference/components/prometheus.exporter.windows.md [hunk 2]
```diff
@@ -271,6 +271,11 @@ Name     | Description | Enabled by default
 
 See the linked documentation on each collector for more information on reported metrics, configuration settings and usage examples.
 
+{{% admonition type="caution" %}}
+Certain collectors will cause the application to crash if they are used and the required infrastructure is not installed.
```
Comment (erikbaranowski) [inline docs/sources/flow/reference/components/prometheus.exporter.windows.md:275]: This is a super helpful addition for folks. It sounds a little awkward to me saying 'the application' but I'll let @clayton-cornell weight in on if the wording makes sense or not.

**Review by erikbaranowski — APPROVED** [review]
LGTM

File: docs/sources/flow/reference/components/prometheus.exporter.windows.md (hunk 2 above)
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/prometheus.exporter.windows.md]: @erikbaranowski @mattdurham The wording is OK as-is. Yes, `the application` is a bit out of place... which application? Agent? Then we should be specific and say this (I should have suggested this in the initial review :-P ). The text also uses passive voice, but that's not a blocker.

We could also consider (in a future edits) swaping `Certain` with `Some` - or is it known which collectors cause the crash? If we know, then we should list them.

These changes aren't critical... we can catch them in a future PR.

**Pull request merged**